src/shapes/Object/InteractiveObject.ts

```diff
@@ -221,11 +232,13 @@ export class InteractiveFabricObject<
       dim = this._calculateCurrentDimensions(transformOptions),
       coords: Record<string, TOCoord> = {};
 
-    this.forEachControl(
-      (control: any, key: string, fabricObject: InteractiveFabricObject) => {
-        coords[key] = control.positionHandler(dim, finalMatrix, fabricObject);
-      }
-    );
+    this.forEachControl((control, key) => {
+      const position = control.positionHandler(dim, finalMatrix, this, control);
+      coords[key] = Object.assign(
```

Comment:
using assign since I think we use it as Point as well

Reply:
i will clarify this with a comment. so it doesn't get swapped with a spread operator